src/main/java/sirius/web/templates/pdf/ImageReplacedElementFactory.java

```diff
@@ -59,44 +56,28 @@ public ReplacedElement createReplacedElement(LayoutContext c,
             return null;
         }
 
+        ReplacedElement fallBackElement = super.createReplacedElement(c, box, uac, cssWidth, cssHeight);
         String nodeName = e.getNodeName();
         if (!TAG_TYPE_IMG.equals(nodeName)) {
-            return super.createReplacedElement(c, box, uac, cssWidth, cssHeight);
+            return fallBackElement;
         }
 
         String src = e.getAttribute(ATTR_SRC);
         if (Strings.isEmpty(src)) {
-            return super.createReplacedElement(c, box, uac, cssWidth, cssHeight);
+            return fallBackElement;
         }
 
         try {
-            Optional<FSImage> image = resolveUri(src, uac, cssWidth, cssHeight);
-
-            if (image.isPresent()) {
-                return new ITextImageElement(image.get());
-            }
+            String protocol = Strings.split(src, "://").getFirst();
+            PdfReplaceHandler handler = findHandler(protocol);
+            AsyncLoadedImageElement imageElement = new AsyncLoadedImageElement(handler, src, cssWidth, cssHeight);
```

Comment:
Maybe we should implement async handling more fine granular? Like only for external http resource resolving?

Reply:
As discussed, I would leave it as it is for now, as we are not only resolving the images here, but also scaling them.